src/catalua.h

```diff
@@ -3,9 +3,67 @@
 #define CATALUA_H
 
 #include "int_id.h"
+#include "enums.h"
+#include "item.h"
 
 #include <string>
 #include <sstream>
+#include <vector>
+#include <list>
+
+#ifdef LUA
+extern "C" {
+#include "lua.h"
+#include "lualib.h"
+#include "lauxlib.h"
+}
+#endif //LUA
+
+enum CallbackArgumentType : int {
+    Undefined = -1,
+    Integer = 0,
+    Number,
+    Double = Number,
+    Float = Number,
+    String,
+    Tripoint,
+    Item
+};
+
+struct CallbackArgument {
+    std::string name = "";
+    CallbackArgumentType type = CallbackArgumentType::Undefined;
+
+    int value_integer = 0;
+    float value_number = 0.0f;
+    std::string value_string = "";
+    tripoint value_tripoint = tripoint( 0, 0, 0 );
+    item value_item = item();
```

Comment:
The ` = item()` is quite pointless. The compiler will automatically invoke the default constructor when your constructor does not initialize the member.

Similar with the string*s* above: a default created `std::string` is already empty, no need to create it from an empty C-String.

And also the `tripoint`: the default constructor of it already sets all components to 0, no need to do this manually.